stackdriver.go

```diff
@@ -155,9 +168,9 @@ type Exporter struct {
 
 // NewExporter creates a new Exporter that implements both stats.Exporter and
 // trace.Exporter.
-func NewExporter(o Options) (*Exporter, error) {
+func NewExporter(ctx context.Context, o Options) (*Exporter, error) {
```

Comment:
I initially had this implemented as a single function that provided context both for initializing the Stackdriver clients, and uploading actual traces/stats (https://github.com/knyar/opencensus-go-exporter-stackdriver/commit/6489d3e).

I then realized that the function also needs to return `CancelFunc` if we'd like this to be used as a way to configure deadlines for API calls, to allow the library to cancel the context [when it's no longer needed](https://github.com/golang/go/blob/27a9b1bbecc0b1aa02d88128ef36904e6f56ec43/src/context/context.go#L443). However, `NewExporter` probably should not be calcelling the provided context, since Stackdriver clients it created might still be using it for underlying connections (even though grpc currently [does not seem](https://github.com/grpc/grpc-go/blob/97da9e087c5cf7a11263637efb410e08c51a7a0f/clientconn.go#L120) to be doing that).

I think there are three ways forward:
1. Don't intend this to be used for configuring deadlines, and have a single `options.NewContext()` function that only returns a context (without `CancelFunc`).
2. Have two separate options: `NewContext() context` (used to create SD clients) and `NewUploadContext() (context, CancelFunc)` (used to upload stats&traces)
3. Add `func NewExporterContext(ctx context.Context, o Options)`, leaving only a single function in options (``NewUploadContext() (context, CancelFunc)`).

I will probably go with option 3, but just wanted to provide you more context (pun intended) in case you have any objections or alternative proposals.

Reply:
I'm happy with option 3. We should deprecate the old non-context function then. 

/cc @rakyll